config_test.go

```diff
@@ -106,3 +107,39 @@ func TestConfigWithInvalidPaths(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigWithMissingAttributes(t *testing.T) {
+	tests := []struct {
+		desc      string
+		cfg       Config
+		expectErr string
+	}{
+		{
+			desc: "missing_level",
+			cfg: Config{
+				Encoding: "json",
+			},
+			expectErr: "missing Level",
+		},
+		{
+			desc: "missing_encoder_time_in_encoder_config",
+			cfg: Config{
+				Level:    NewAtomicLevelAt(zapcore.InfoLevel),
+				Encoding: "json",
+				EncoderConfig: zapcore.EncoderConfig{
+					MessageKey: "msg",
+					TimeKey:    "ts",
+				},
+			},
+			expectErr: "missing EncodeTime in EncoderConfig",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			cfg := tt.cfg
+			_, err := cfg.Build()
```

Comment:
err.Error on the following line will panic if we don't first verify that the error is non-nil.
We should use require.Error here.

```suggestion
			_, err := cfg.Build()
			require.Error(t, err)
```